llvm/lib/IR/Function.cpp

```diff
@@ -378,6 +380,40 @@ Function *Function::createWithDefaultAttr(FunctionType *Ty,
   }
   if (M->getModuleFlag("function_return_thunk_extern"))
     B.addAttribute(Attribute::FnRetThunkExtern);
+
+  Triple T(M->getTargetTriple());
+
+  auto Profile = llvm::ARM::parseArchProfile(T.getArchName());
+  if ((T.isArmT32() && Profile == llvm::ARM::ProfileKind::M) || T.isAArch64()) {
```

Comment:
Perhaps you can avoid bringing in Triple.h and ARMTargetParser if we just always check for these attributes (even if they are arch specific?  I guess I'm curious about the M profile check here.  If we ran this on non-m-profile arm32, would the module attributes even ever exist?

Reply:
That won't happen because the fronted won't add the flag, even it they present the backend won't do anything with them.